uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1030.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+
+trait CRM_HRLeaveAndAbsences_Upgrader_Step_1030 {
+  
+  /**
+   * Disables or deletes existing "sickness reason" option values
+   * and added new ones
+   *
+   * @return bool
+   */
+  public function upgrade_1030() {
+    $oldSicknessReasons = civicrm_api3('OptionValue', 'get', [
+      'return' => ['name'],
+      'option_group_id' => 'hrleaveandabsences_sickness_reason',
+    ]);
+  
+    $result = $this->up1030_getLeaveRequestTotal();
+    if ($result == 0) {
+      $this->up1030_deleteOldSicknessReasons($oldSicknessReasons['values']);
+    }
+    else {
+      $this->up1030_disableOldSicknessReasons($oldSicknessReasons['values']);
+    }
+    
+    $this->up1030_addNewSicknessReasons();
+    
+    return TRUE;
+  }
+  
+  /**
+   * Counts the total number of leave requests related to sickness reasons
+   *
+   * @return int
+   */
+  private function up1030_getLeaveRequestTotal() {
```

Comment:
I don't think the method name is very descriptive here, sounds like the total number of leave requests, but actually it's the total number that have a sickness reason

Reply:
I will rename. Thanks